qml/DatePickerDialog.qml

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (C) Olivier Le Doeuff 2019
+ * Contact: olivier.ldff@gmail.com
+ */
+
+// Qt
+import QtQuick 2.12
+
+// Qaterial
+import Qaterial 1.0 as Qaterial
+
+Qaterial.ModalDialog
+{
+  id: root
+
+  property int month: new Date().getMonth()
+  property int year: new Date().getFullYear()
+
+  property date from: new Date(1900, 0, 1)
+  property date to: new Date(2099, 11, 31)
+  property date date
+
+  property bool monthGridActive: false
+  property bool yearGridActive: false
+  property bool calendarViewActive: true
+
+  padding : 0
+
+  standardButtons: Dialog.Ok | Dialog.Cancel
+
+  signal accepted (date date)
+
+  header: Qaterial.CalendarDateDisplay
+  {
+    function printDate()
+    {
+      var m = Qaterial.Calendar.monthToString(root.date.getMonth())
+      var date = root.date.getDate()
+      var day = Qaterial.Calendar.dayToString(root.date.getDay())
+
+      var y = root.date.getFullYear()
+      var dateString = `${day[0]}${day[1]}${day[2]}, ${m[0]}${m[1]}${m[2]} ${date}, ${y}`
+
+
+      return dateString
+    }
+    width: 500
+
+    text: printDate()
+  }
+
+  contentItem: Item
+  {
+    implicitHeight : _calendar.implicitHeight
+    implicitWidth: _calendar.implicitWidth
+    Loader
```

Comment:
You should use one Loader and 3 Component

Then loader could be the contentItem. Width and Height should be set in the loader